omero/developers/Python.rst

```diff
@@ -310,6 +341,31 @@ Read data
                 well.getImage(index).getName(),\
                 well.getImage(index).getId()
 
+-  **List all annotations on an object. Get text from tags**
+
+::
+
+    for ann in project.listAnnotations():
+        print ann.getId(), ann.OMERO_TYPE,
+        print " added by ", ann.link.getDetails().getOwner().getOmeName()
```

Comment:
sometimes properties are hidden behind "getDetails()", in other cases, they are not. Is there a way to get my head around this ?

Reply:
don't know the best way to clarify this. Details are not specific to any particular object. All objects have details and owner (except Group and Experimenter) but other properties e.g. name or strokeWidth are specific to certain objects.

Reply:
Can check https://docs.openmicroscopy.org/omero/5.5.1/developers/Model/EveryObject.html to see if things have details.